src/shapes.ts

```diff
@@ -0,0 +1,270 @@
+/**
+ * @packageDocumentation
+ * @module settings
+ *
+ * This file consists of shapes supported by chemiscope via 3dmol
+ *
+ * To add a new shape type, extend the `Shape` class and define the
+ * inherited functions (`validateParams` and `outputTo3Dmol`).
+ * `validateParams` should check a `ShapeData` object to see that
+ * it has the correct parameters for the given type. `outputTo3Dmol`
+ * should take a color and output a `CustomShapeSpec` (https://3dmol.csb.pitt.edu/doc/types.html#CustomShapeSpec)
+ * object. You will also need to add any parameters for your shape
+ * to ShapeData.
+ *
+ * To add a new shape, you will also need to add code to `structure/viewer.ts`.
+ */
+
+import { default as $3Dmol } from '3dmol';
+import { Quaternion } from '3dmol';
+
+// Just an interface to enforce XYZ-type coordinates
+export interface XYZ {
+    x: number;
+    y: number;
+    z: number;
+}
+
+// A base class for storing shape data in ase files
+export interface ShapeData {
+    radius?: number;
+    semiaxes?: [number, number, number];
+    orientation?: [number, number, number, number];
+    kind: string;
+}
```

Comment:
```suggestion
export interface SphereData {
    radius?: number;
    kind: "sphere";
}

export interface EllipsoidData {
    semiaxes?: [number, number, number];
    orientation?: [number, number, number, number];
    kind: "ellipsoid";
}

export ShapeData = SphereData | EllipsoidData;
```

This should work with TypeScript type system and reflects what we want better

Reply:
This does not work, and throws several errors:

```[tsl] ERROR in /path/to/chemiscope/src/shapes.ts(40,30)
      TS2693: 'SphereData' only refers to a type, but is being used as a value here.
 @ ./src/structure/viewer.ts 9:0-46 682:23-47 683:41-50 687:23-44 688:38-44

ERROR in /path/to/chemiscope/src/shapes.ts
./src/shapes.ts 40:42-55
[tsl] ERROR in /path/to/chemiscope/src/shapes.ts(40,43)
      TS2693: 'EllipsoidData' only refers to a type, but is being used as a value here.
 @ ./src/structure/viewer.ts 9:0-46 682:23-47 683:41-50 687:23-44 688:38-44
```

Reply:
Looks like the right syntax is `export type ShapeData = SphereData | EllipsoidData;` (maybe no export?)

https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#union-types

Reply:
Done!